server/datastore/mysql/policies.go

```diff
@@ -736,6 +745,15 @@ func (ds *Datastore) ListMergedTeamPolicies(ctx context.Context, teamID uint, op
 }
 
 func (ds *Datastore) DeleteTeamPolicies(ctx context.Context, teamID uint, ids []uint) ([]uint, error) {
+	for _, id := range ids {
+		if err := ds.deletePendingSoftwareInstallsForPolicy(ctx, id); err != nil {
```

Comment:
Let's pass `teamID *uint` as argument to `deletePendingSoftwareInstallsForPolicy` and `deletePendingHostScriptExecutionsForPolicy` (and pass `nil` on `DeleteGlobalPolicies`) so that we can be sure the passed and deleted `ids` belong to the team/global.